sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGenerateExec.scala

```diff
@@ -53,12 +53,54 @@ class GpuGenerateExecSparkPlanMeta(
   }
 
   override def convertToGpu(): GpuExec = {
-    GpuGenerateExec(
-      childExprs.head.convertToGpu().asInstanceOf[GpuGenerator],
-      gen.requiredChildOutput,
-      gen.outer,
-      gen.generatorOutput,
-      childPlans.head.convertIfNeeded())
+    // if child expression contains Stack, use GpuExpandExec instead
+    if (childExprs.exists(_.isInstanceOf[GpuStackMeta])) {
```

Comment:
Only one expression is allowed by Spark, so

```suggestion
    childExprs.head match {
      case stackMeta: GpuStackMeta =>
        ......
      case genMeta =>
        val generator = genMeta.convertToGpu().asInstanceOf[GpuGenerator]
        ......
    }
      
```